Changes for lazy loading module on client side

MrSpark2591 opened the pull request:
This might be breaking changes as this will remove AWS object from global scope and we have to import it as we go.

MrSpark2591 committed:
made for client side only

MrSpark2591 committed:
changes for lazy loading

mitar commented on client-tests.js:
```diff
@@ -1,16 +1,3 @@
-  test.isTrue(isDefined, "AWS is not defined");
-  test.isTrue(Package['peerlibrary:aws-sdk'].AWS, "Package.peerlibrary:aws-sdk.AWS is not defined");
-
-  var s3 = new AWS.S3();
```
Could you keep these tests?

MrSpark2591 replied:
For lazy loading it shouldn't get imported. That's why i have removed these test. Also i was unable to find out how to test out lazy loading package. Please guide me.

mitar commented on client-tests.js:
```diff
@@ -1,16 +1,3 @@
-  test.isTrue(s3.upload);
-});
+Tinytest.add('aws-sdk', function(test) {
+  test.isFalse(Package['peerlibrary:aws-sdk'].AWS, "Package.peerlibrary:aws-sdk.AWS is not defined");
```
Change `not defined` to `defined`.

MrSpark2591 replied:
This will be resolved if we are able to find out a way to test lazy loaded package.

mitar commented on .versions:
```diff
@@ -1,11 +1,11 @@
 allow-deny@1.1.0
```
Don't modify this file. It is modified when I publish the package.

MrSpark2591 replied:
yes sure i will revert my changes.

mitar commented on server-tests.js:
```diff
@@ -1,18 +1,10 @@
   test.isTrue(Package['peerlibrary:aws-sdk'].AWS, "Package.peerlibrary:aws-sdk.AWS is not defined");
 
-  var s3 = new AWS.S3();
+  var s3 = new Package['peerlibrary:aws-sdk'].AWS.S3();
```
Why not importing the package?

MrSpark2591 replied:
I will import it and submit PR.

MrSpark2591 committed:
changes as per review

MrSpark2591 commented:
@mitar I have reverted version bumping and changed PR as per your input. Please publish package once this branch has been merged with version bumping.

MrSpark2591 commented:
@mitar Is it possible to publish the package so that I can change my dependency to this package instead of my forked one?

mitar commented:
Published as `2.598.0_1`. In fact there was no easy way to demonstrate the backwards incompatibility of this change. :-(
